src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -4169,6 +4201,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
             _DeploymentComputeNativeManifestInfo;
             _DeploymentComputeClickOnceManifestInfo;
             ResolveComReferences;
+            GetComAssembliesMetadata;
```

Comment:
The new task should run only in design time builds. It is currently missing such a condition and it runs in regular builds as well, which is not desired.

Reply:
Also, it appears to be quite slow. I built a trivial project on the command line and got this:

![image](https://github.com/dotnet/msbuild/assets/12206368/fc57df35-10da-4005-b0ef-8bfcb37a198f)

Reply:
Assuming the new project system doesn't need this task and it's really only for CSProj, I don't think the way it's currently done is correct. Instead of adding it to `ResolveReferencesDependsOn` in this file, do you think it would be possible for CSProj to explicitly request the target to run?

Side question: Does the new project system also extract this metadata? If so, what is the mechanism?

Also, why is it added as a `GenerateApplicationManifest` dependency here?